src/modalities/utils/mfu.py

```diff
@@ -37,53 +39,59 @@ def _get_theoretical_gpu_peak_performance_single(precision: torch.dtype, gpu_typ
         return None
 
 
-def get_theoretical_gpu_peak_performance(model: FSDP, world_size: int) -> Optional[Number]:
+def get_theoretical_gpu_peak_performance(model: FSDPX, world_size: int) -> Optional[Number]:
     """
-    Calculates the accummulated theoretical peak performance based on all GPUs, i.e.,
+    Calculates the accumulated theoretical peak performance based on all GPUs, i.e.,
       #GPU=world_size, in units FLOPs / s for given gpu type.
 
     Args:
-        model (FSDP): The model for which to calculate the theoretical peak performance.
+        model (FSDPX): The model for which to calculate the theoretical peak performance.
         world_size (int): The number of GPUs used in parallel.
 
     Returns:
-        (Number, optional): The accummulated theoretical peak performance of all GPUs,
+        (Number, optional): The accumulated theoretical peak performance of all GPUs,
           or None if it cannot be calculated.
     """
     if torch.cuda.is_available() and torch.cuda.device_count() > 0:
-        precision = model.mixed_precision.param_dtype
-        if model.mixed_precision.reduce_dtype != precision or model.mixed_precision.buffer_dtype != precision:
-            warnings.warn(f"Could not get theoretical GPU peak performance for mixed precision type = {precision}.")
+        if isinstance(model, FSDP1):
+            precision = model.mixed_precision.param_dtype
+            if model.mixed_precision.reduce_dtype != precision or model.mixed_precision.buffer_dtype != precision:
+                warnings.warn(f"Could not get theoretical GPU peak performance for mixed precision type = {precision}.")
+                return None
+        elif isinstance(model, FSDP2):
+            warnings.warn("MFU is computed based on the assumption that bf16 precision is used.")
+            precision = torch.bfloat16
```

Comment:
We also plan to use FP8.
TODO: Create an issue.

Reply:
done